packages/generate-arg-types/src/cli.ts

```diff
@@ -54,50 +55,97 @@ if (componentFiles.length === 0) {
+      customDescriptionsByDir[customDescriptionsDir];
+
+    if (customDescriptions == null) {
+      try {
+        const { propsDescriptions } = await import(
+          path.join(customDescriptionsDir, "props-descriptions.ts")
+        );
+        customDescriptionsByDir[customDescriptionsDir] = propsDescriptions;
+        customDescriptions = propsDescriptions;
+      } catch (error) {
+        customDescriptions = {};
+      }
+    }
 
-  const componentDocs = tsConfigParser.parse(filePath);
+    const generatedFile = `${path.basename(filePath, ".tsx")}.props.ts`;
+    const generatedPath = path.join(generatedDir, generatedFile);
 
-  if (componentDocs.length === 0) {
-    console.error(`No propTypes found for ${filePath}`);
-    continue;
-  }
+    const componentDocs = tsConfigParser.parse(filePath);
 
-  let fileContent = `import type { PropMeta } from "@webstudio-is/generate-arg-types";\n`;
+    if (componentDocs.length === 0) {
+      console.error(`No propTypes found for ${filePath}`);
+      continue;
+    }
 
-  if (componentDocs.length === 1) {
-    const argTypes = propsToArgTypes(
-      componentDocs[0].props,
-      cliArgs.values.exclude ?? []
-    );
-
-    fileContent = `${fileContent}
-      export const props: Record<string, PropMeta> = ${JSON.stringify(
-        argTypes
-      )}`;
-  } else {
-    for (const componentDoc of componentDocs) {
-      const componentName = componentDoc.displayName;
+    let fileContent = `import type { PropMeta } from "@webstudio-is/generate-arg-types";\n`;
 
+    if (componentDocs.length === 1) {
       const argTypes = propsToArgTypes(
-        componentDoc.props,
+        componentDocs[0].props,
         cliArgs.values.exclude ?? []
       );
 
+      const componentName = componentDocs[0].displayName;
+
+      addDescriptions(
+        componentName,
+        argTypes,
+        customDescriptions[componentName]
+      );
+
       fileContent = `${fileContent}
+      export const props: Record<string, PropMeta> = ${JSON.stringify(
+        argTypes
+      )}`;
+    } else {
+      for (const componentDoc of componentDocs) {
+        const argTypes = propsToArgTypes(
+          componentDoc.props,
+          cliArgs.values.exclude ?? []
+        );
+
+        const componentName = componentDoc.displayName;
+
+        addDescriptions(
+          componentName,
+          argTypes,
+          customDescriptions[componentName]
+        );
+
+        fileContent = `${fileContent}
         export const props${componentName}: Record<string, PropMeta> = ${JSON.stringify(
           argTypes
         )}`;
+      }
     }
-  }
 
-  mkdirSync(generatedDir, { recursive: true });
-  writeFileSync(generatedPath, fileContent);
+    mkdirSync(generatedDir, { recursive: true });
+    writeFileSync(generatedPath, fileContent);
 
-  console.log(`Done generating argTypes for ${generatedPath}`);
-}
+    console.log(`Done generating argTypes for ${generatedPath}`);
+  }
+})();
```

Comment:
Why not top level await?

Reply:
Error or warning? Caz we don't care about cli for end user.

Reply:
Error